### PR TITLE
define DigitalObject#==, let it be used implicitly in test comparisons

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made? 🤔

* pro: this makes the tests more concise and adds a possibly useful equality checking method
* con: i'm not sure that this is actually captures the desired semantics of `DigitalObject#==`.  also, only the tests use it at present.  i think the comparison works for the tests, but for example, should it rely on the default instance-equivalence equality check for `Batch`, or should `Batch` be compared for semantic equality between possibly different instances instead?

## How was this change tested? 🤨

unit tests

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to shared file systems or how it uses APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Comment by ndushay** [comment]
@jmartin-sul who o you believe can answer this question?  Should it be a post standup topic?

**Review by ndushay — CHANGES_REQUESTED** [review]
would want method in spec_helper if it's only used for tests.

**Comment by peetucket** [comment]
Created spec_helper method and used in the tests.  Removed equals method in digital object class.

**Review by ndushay — APPROVED** [review]

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -72,3 +72,12 @@
   # particularly slow.
   # config.profile_examples = 10
 end
+
+# iterate over an array of digital objects, and verify they have equivalent attributes to a second array of digital objects
+def objects_equivalent?(objects1, objects2)
```
[the remark this answers is not in the document]
Comment (peetucket) [reply spec/spec_helper.rb]: renamed the method

**Comment by jmartin-sul** [comment]
thanks @peetucket!

**Pull request merged**